required/latex-lab/latex-lab-everymath.dtx

```diff
@@ -0,0 +1,347 @@
+%    \end{macrocode}
+%
+%    \begin{macrocode}
+%<*kernel>
+%    \end{macrocode}
+%
+% \subsection{File declaration}
+%    \begin{macrocode}
+\ProvidesFile{latex-lab-everymath.ltx}
+        [2022-12-09 v0.1a Grab all the math(s)]
+%    \end{macrocode}
+%
+%    \begin{macrocode}
+\ExplSyntaxOn
+%    \end{macrocode}
+%
+% \subsection{Setup}
+%
+% Loading \pkg{amsmath} is an absolute requirement: this avoids needing to
+% have conditional definitions and deals with how to define \cs{[}/\cs{]}
+% neatly.
+%    \begin{macrocode}
+\AddToHook { begindocument / before }
+  { \RequirePackage { amsmath } }
+%    \end{macrocode} 
+%
+% \subsection{Data structures}
+%
+% \begin{variable}{\l_@@_cmd_bool}
+%   Tracks whether we are dealing with \LaTeX{}'s math mode (which is
+%   referred to here as \texttt{cmd}).
+%    \begin{macrocode}
+\bool_new:N \l_@@_cmd_bool
+%    \end{macrocode}
+% \end{variable}
+%
+% \begin{variable}{\l_@@_inner_bool}
+%   Tracks if we are inside a top-level inline math context.
+%    \begin{macrocode}
+\bool_new:N \l_@@_inner_bool
+%    \end{macrocode}
+% \end{variable}
+%
+% \begin{variable}{\l_@@_display_bool}
+%   Tracks display style (cases where \cs{everydisplay} is triggered).
+%    \begin{macrocode}
+\bool_new:N \l_@@_display_bool
+%    \end{macrocode}
+% \end{variable}
+%
+% \begin{variable}{\l_@@_env_bool}
+%   Tracks math environments which deal with grabbing themselves.
+%    \begin{macrocode}
+\bool_new:N \l_@@_env_bool
+%    \end{macrocode}
+% \end{variable}
+%
+% \subsection{Interface commands}
+%
+% \begin{macro}{\@@_process:n, \@@_process_aux:n}
+%   A no-op place-holder; the internal wrapper means that it does not need to
+%   be concerned with internals.
+%    \begin{macrocode}
+\cs_new_protected:Npn \@@_process:n #1
+  {
+    \legacy_if:nF { measuring@ }
+      { \@@_process_aux:n {#1} }
+  }
+\cs_new_protected:Npn \@@_process_aux:n #1 { }
+%    \end{macrocode}
+% \end{macro}
+%
+% \begin{macro}{\math_processor:n}
+%   A simple installer
+%    \begin{macrocode}
+\cs_new_protected:Npn \math_processor:n #1
+  { \cs_set_protected:Npn \@@_process_aux:n ##1 {#1} }
+%    \end{macrocode}
+% \end{macro}
+%
+% \subsection{Content grabbing}
+%
+% \begin{macro}{\@@_grab_inline_dollar:w}
+%   Grab up to a single |$|, for inline math mode.
+%    \begin{macrocode}
+\cs_new_protected:Npn \@@_grab_inline_dollar:w % $
+  #1 $
+  {
+    \@@_process:n {#1} % $
+    #1 $
+  }
+%    \end{macrocode}
+% \end{macro}
+% \begin{macro}{\@@_grab_inline_cmd:w}
+%   The same but for the \LaTeX{} structure.
+%    \begin{macrocode}
+\cs_new_protected:Npn \@@_grab_inline_cmd:w % \(
+  #1 \)
+  {
+    \@@_process:n {#1} % \(
+    #1 \)
+  }
+%    \end{macrocode}
+% \end{macro}
+%
+% \begin{macro}{\@@_grab_dollardollar:w}
+%   And for the classical \TeX{} display structure.
+%    \begin{macrocode}
+\cs_new_protected:Npn \@@_grab_dollardollar:w % $$
+  #1 $$
+  {
+    \@@_process:n {#1} % $$
+    #1 $$
+  }
+%    \end{macrocode}
+% \end{macro}
```

Comment:
suggesting to use `(+)b` is fine, but we should then document that with an example or two how to change an existing definition. amsmath envs can be hidden and are occasionally if you use the command form instead of the environment form.

Reply:
@FrankMittelbach  If we move the current hook-based approach to simply modifying `\<ams-thing>`, then there's no change compared with the normal `amsmath` situation here (as for `amsmath` stuff we don't actually need to do any collecting anyway). Want me to do that?